gh-122196: Clarify, reorder language for TestCase(object)

mavaddat opened the pull request:
The former docstring inappropriately uses the term 'fixture' before introducing or defining the term. Therefore, I have re-arranged the description so that the definition comes before the usage. Additional semantic changes made to be clearer, more understandable, and linear.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- gh-issue-number: gh-122196 -->
* Issue: gh-122196
<!-- /gh-issue-number -->

mavaddat committed:
Clarify, reorder language for TestCase(object)

The former docstring inappropriately uses the term 'fixture' before introducing or defining the term. Therefore, I have re-arranged the description so that the definition comes before the usage. Additional semantic changes made to be clearer, more understandable, and linear.

mavaddat commented on Lib/unittest/case.py:
```diff
@@ -331,8 +333,8 @@ class TestCase(object):
-        objects used in assert methods) will be printed on failure in *addition*
-        to any explicit message passed.
+        objects used in assert methods) will be printed on failure in
+        *addition* to any explicit message passed.
```
To conform with [PEP 8 E501](https://www.flake8rules.com/rules/E501.html).

> Line lengths are recommended to be no greater than 79 characters. The reasoning for this comes from PEP8 itself:
> 

`Limiting the required editor window width makes it possible to have several files open side-by-side, and works well when using code review tools that present the two versions in adjacent columns.`

> 
> It is fairly common for developers, especially those in closed-source projects, to change the maximum line length to 100 or 120 characters.

MaxwellDupre reviewed (approved):
Looks ok

CoolCat467 reviewed (approved):
Makes sense

mavaddat committed:
Merge branch 'main' into master

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

mavaddat committed:
Merge branch 'main' into master

AA-Turner commented on Lib/unittest/case.py:
```diff
@@ -385,8 +387,8 @@ class TestCase(object):
-        objects used in assert methods) will be printed on failure in *addition*
-        to any explicit message passed.
+        objects used in assert methods) will be printed on failure in
+        *addition* to any explicit message passed.
```
Unrelated to the main change (and also 80 chars is fine)
```suggestion
        objects used in assert methods) will be printed on failure in *addition*
        to any explicit message passed.
```

AA-Turner commented on Lib/unittest/case.py:
```diff
@@ -362,18 +362,20 @@ def __iter__(self):
+    Testing code may be placed in subclass method named 'runTest';
+    alternatively, invoke unittest.main() from the subclass's "__name__ guard"
+    section to run all methods defined in the test class. To achieve more
+    precise subclass test invocation see below (viz., constructor kwarg).
```
The ``runTest`` form is far rarer, should we cover it second?

mavaddat replied:
I rewrote the docs according to your suggestion.

AA-Turner commented on Lib/unittest/case.py:
```diff
@@ -362,18 +362,20 @@ def __iter__(self):
+    If the fixture will be used for several distinct test cases, create as
+    many test methods as needed in the TestCase subclass, then specify a
+    method to execute by providing the method name in the 'methodName'
+    kwarg of the instance constructor.
```
```suggestion
    parameter of the instance constructor.
```

AA-Turner commented on Misc/NEWS.d/next/Documentation/2024-07-23-20-18-31.gh-issue-122196.pYwrvo.rst:
You can delete this file, the bot complains until the label is added.

mavaddat committed:
Merge branch 'main' into master

mavaddat committed:
Update Lib/unittest/case.py

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

mavaddat committed:
Update Lib/unittest/case.py

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

mavaddat committed:
Update case.py

Re-order recommendations as suggested by https://github.com/python/cpython/pull/24465#discussion_r1688897655
